ethers-middleware/src/nonce_manager.rs

```diff
@@ -142,7 +155,19 @@ where
         tracing::debug!(?nonce, "Sending transaction");
         match self.inner.send_transaction(tx.clone(), block).await {
             Ok(pending_tx) => {
+                let new_txs_since_resync = self.txs_since_resync.fetch_add(1, Ordering::SeqCst);
                 tracing::debug!(?nonce, "Sent transaction");
+                let tx_count_for_resync = self.get_tx_count_for_resync();
+                if new_txs_since_resync >= tx_count_for_resync {
+                    let onchain_nonce = self.get_transaction_count(self.address, block).await?;
+                    self.nonce.store(onchain_nonce.as_u64(), Ordering::SeqCst);
+                    self.txs_since_resync.store(0, Ordering::SeqCst);
+                    tracing::debug!(?nonce, "Resynced internal nonce with onchain nonce");
+                } else {
+                    self.txs_since_resync.store(new_txs_since_resync, Ordering::SeqCst);
```

Comment:
nit: iiuc `fetch_add` stores the new value https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_add so we don't need to store again

or maybe you'd prefer to have 158 not store and just fetch?

Reply:
sweet didn't know that, removed this line

Reply:
actually the behavior I want is not to receive the old value in `new_txs_since_resync` but the new one, so I'll just `fetch` there and `store` here